ddtrace/contrib/pylons/middleware.py

```diff
@@ -41,17 +41,22 @@ def _start_response(status, *args, **kwargs):
             except Exception as e:
                 # "unexpected errors"
                 # exc_info set by __exit__ on current tracer
+
+                # e.code can either be a string or an int
                 code = getattr(e, 'code', 500)
-                if isinstance(code, int) and code >= 100 and code < 600:
-                    span.set_tag(http.STATUS_CODE, code)
-                else:
-                    span.set_tag(http.STATUS_CODE, 500)
+                try:
+                    code = int(code)
+                    if not 100 <= code < 600:
+                        code = 500
+                except:
```

Comment:
I think you can handle only the `ValueError` otherwise you're catching all possible exceptions.